rust/lancedb/src/index/vector.rs

```diff
@@ -46,17 +47,20 @@ impl VectorIndex {
     }
 }
 
+#[skip_serializing_none]
 #[derive(Debug, Deserialize)]
 pub struct VectorIndexMetadata {
-    pub metric_type: String,
-    pub index_type: String,
+    pub metric_type: Option<String>,
```

Comment:
why do we need to change this to optional? There are index that do not have these fields?

Reply:
correct, scalar index stats does not have this field.

Reply:
But this struct is called `VectorIndexMetadata`. Do we want to create a new `ScalarIndexMetadata`, or consolidate into one `IndexMetadata`?

Reply:
Probably this should not be called VectorIndexStatistics if it applies to scalar indexes too.

It would definitely be cleaner to have separate ScalarIndexMetadata and VectorIndexMetadata types instead of using optional fields. Is that feasible?

Reply:
Yeah, it should be renamed to `IndexStatistics`
we plan to use createIndex for both vector and scalar index in the future so prob it is a good idea to be unified as `IndexMetadata`.